packages/web-ui/src/ds/molecules/Chat/Message/index.tsx

```diff
@@ -224,91 +200,160 @@ const Content = ({
+    const codeContent = isAgentResponse
+      ? JSON.stringify(value.args, null, 2)
+      : `${value.toolName}(${JSON.stringify(value.args, null, 2)})`
+
+    return (
+      <div key={`${index}`} className='py-2 w-full'>
+        <div className='overflow-hidden rounded-xl w-full flex-col'>
+          <div
+            className={cn(
+              'flex w-full px-2 py-0.5 items-center gap-2 justify-between',
+              headerBgColor,
+              { 'py-1': isAgentResponse },
+            )}
+          >
+            <div className='flex items-center gap-1.5'>
+              <Icon name={headerIcon} color={headerFgColor} />
+              <Text.H6 noWrap color={headerFgColor}>
+                {headerLabel}
+              </Text.H6>
+            </div>
+            {!isAgentResponse && (
+              <TruncatedTooltip content={value.toolCallId}>
+                <Text.H6 color='warningMutedForeground' ellipsis>
+                  {value.toolCallId}
+                </Text.H6>
+              </TruncatedTooltip>
+            )}
           </div>
+          <CodeBlock language={codeLanguage}>{codeContent}</CodeBlock>
         </div>
-      )
+      </div>
+    )
+  }
 
-    case ContentType.toolResult:
-      return (
-        <div key={`${index}`} className='py-2 w-full'>
-          <div className='overflow-hidden rounded-xl w-full'>
-            <CodeBlock language='json'>
-              {JSON.stringify(value as ToolContent, null, 2)}
-            </CodeBlock>
+  if (value.type === ContentType.toolResult) {
+    return (
+      <div key={`${index}`} className='py-2 w-full'>
+        <div className='overflow-hidden rounded-xl w-full'>
+          <div
+            className={cn(
+              'flex w-full px-2 py-0.5 items-center gap-2 justify-between',
+              {
+                'bg-muted': !value.isError,
+                'bg-destructive': value.isError,
+              },
+            )}
+          >
+            <div className='flex items-center gap-1.5'>
+              <Icon
+                name={'terminal'}
+                color={
+                  value.isError ? 'destructiveForeground' : 'foregroundMuted'
+                }
+              />
+              <Text.H6
+                noWrap
+                color={
+                  value.isError ? 'destructiveForeground' : 'foregroundMuted'
+                }
+              >
+                {value.toolName}
+              </Text.H6>
+            </div>
+            <TruncatedTooltip content={value.toolCallId}>
+              <Text.H6
+                color={
+                  value.isError ? 'destructiveForeground' : 'foregroundMuted'
+                }
+                ellipsis
+              >
+                {value.toolCallId}
+              </Text.H6>
+            </TruncatedTooltip>
           </div>
+          <CodeBlock language='json'>
+            {JSON.stringify(value.result, null, 2)}
+          </CodeBlock>
```

Comment:
This switch case is super big. Extract both tool request and tool response in other components